azure/src/createAzureClient.ts

```diff
@@ -283,3 +297,75 @@ class AllowInsecureConnectionPolicy implements PipelinePolicy {
         return await next(request);
     }
 }
+
+/**
+ * Resolve a default scope from the Resource Manager endpoint or any provided endpoint.
+ * Example: https://management.azure.com/ -> https://management.azure.com/.default
+ */
+function getDefaultScopeFromEndpoint(endpoint?: string): string {
+    let base = endpoint ?? 'https://management.azure.com/';
+    base = base.replace(/\/+$/, '');
+    return `${base}/.default`;
+}
+
+/**
+ * Try to extract an OAuth scope or resource from the raw WWW-Authenticate header.
+ * Falls back to deriving a scope from the endpoint if not present.
+ */
+function resolveScopesFromChallengeOrEndpoint(rawHeader: string, endpoint?: string): string[] {
+    const scopeMatch = /(?:^|[,\s])scope="([^"]+)"/i.exec(rawHeader);
+    if (scopeMatch && scopeMatch[1]) {
+        return [scopeMatch[1]];
+    }
+
+    const resourceMatch = /(?:^|[,\s])resource(?:_id)?=\"?([^\",\s]+)\"?/i.exec(rawHeader);
```

Comment:
Any reason why in this regex, double quotes are escaped, but not in the above one?